client/src/components/assessments/AssessmentResultsContainer.js

```diff
@@ -20,10 +21,24 @@ const AssessmentResultsContainer = ({
   const assessmentsTypes = Object.keys(entity.getAssessmentsConfig())
   return (
     <div ref={contRef}>
-      {assessmentsTypes.map(
-        assessmentsType =>
-          PERIOD_FACTORIES[assessmentsType] && (
-            <AssessmentResultsTable
+      {assessmentsTypes.map(assessmentsType =>
+        PERIOD_FACTORIES[assessmentsType] ? (
+          <AssessmentResultsTable
+            key={assessmentsType}
+            style={{ flex: "0 0 100%" }}
+            entity={entity}
+            entityType={entityType}
+            subEntities={subEntities}
+            periodsDetails={{
+              recurrence: assessmentsType,
+              numberOfPeriods
+            }}
+            canAddAssessment={canAddAssessment}
+            onUpdateAssessment={onUpdateAssessment}
+          />
+        ) : (
+          assessmentsType === "ondemand" && (
```

Comment:
Should `"ondemand"` be defined as a global constant somewhere, e.g. in `periodUtils.js`?